pygmt/src/triangulate.py

```diff
@@ -0,0 +1,384 @@
+        """
+        with Session() as lib:
+            # Choose how data will be passed into the module
+            table_context = lib.virtualfile_from_data(
+                check_kind="vector", data=data, x=x, y=y, z=z, required_z=False
+            )
+            with table_context as infile:
+                if "G" not in kwargs:  # table output if outgrid is unset
+                    kwargs.update({">": outfile})
+                else:  # NetCDF or xarray.DataArray output if outgrid is set
+                    outgrid = kwargs["G"]
+                arg_str = " ".join([infile, build_arg_string(kwargs)])
+                lib.call_module(module="triangulate", args=arg_str)
+
+        if output_type == "file":
+            return None
+        if output_type == "xarray":
+            return load_dataarray(outgrid)
+
+        result = pd.read_csv(outfile, sep="\t", header=None)
+        if output_type == "numpy":
+            return result.to_numpy()
+        return result
+
+    @staticmethod
+    @fmt_docstring
+    @kwargs_to_strings(R="sequence")
+    def regular_grid(  # pylint: disable=too-many-arguments,too-many-locals
+        data=None,
+        x=None,
+        y=None,
+        z=None,
+        outgrid=None,
+        spacing=None,
+        projection=None,
+        region=None,
+        verbose=None,
+        binary=None,
+        nodata=None,
+        find=None,
+        coltypes=None,
+        header=None,
+        incols=None,
+        registration=None,
+        skiprows=None,
+        wrap=None,
+        **kwargs,
+    ):
+        """
+        Delaunay triangle based gridding of Cartesian data.
+
+        Reads in x,y[,z] data and performs Delaunay triangulation, i.e., it
+        finds how the points should be connected to give the most equilateral
+        triangulation possible. If a map projection (give *region* and
+        *projection*) is chosen then it is applied before the triangulation is
+        calculated. By setting **outgrid** and **spacing**, a grid will be
+        calculated based on the surface defined by the planar triangles. The
+        actual algorithm used in the triangulations is either that of Watson
+        [1982] or Shewchuk [1996] [Default] (if installed; type
+        **gmt get GMT_TRIANGULATE** on the command line to see which method is
+        selected). This choice is made during the GMT installation.
+        Furthermore, if the Shewchuk algorithm is installed then you can also
+        perform the calculation of Voronoi polygons and optionally grid your
+        data via the natural nearest neighbor algorithm. **Note**: For
+        geographic data with global or very large extent you should consider
+        :gmt-docs:`sphtriangulate <sphtriangulate.html>` instead since
+        **triangulate** is a Cartesian or small-geographic area operator and is
+        unaware of periodic or polar boundary conditions.
+
+        Must provide either ``data`` or ``x``, ``y``, and ``z``.
```

Comment:
```suggestion
        Must provide either ``data`` or ``x``, ``y``, and ``z``.
        
        Must provide ``region`` and ``spacing``.
```
Region and spacing are required unless **-C** is used, which I think should go in a separate method.